Add multicontainer and regular pnnl dopf

josephmckinsey opened the pull request:
~Currently the multicontainer download fails~ Fixed.

Tylores commented:
- cvxpy is not included in requirements.txt so oedisi build fails for the regular scenario.
- the scenario file includes an "existing_feeder_file" which doesn't have a fallback by the feeder federate if it doesn't exist. 
- docker compose up doesn't work from the project directory and needs to be called from build to work. It should default to build and only need specification if you build to a different folder. 
- There is a PORT environment variable that causes the oedisi broker container to fail.
<img width="839" alt="oedisi-multi-error" src="https://github.com/openEDI/oedisi-example/assets/20763042/56bcfbba-41ea-4a11-a50f-85b0b3d65828">

josephmckinsey commented:
> * cvxpy is not included in requirements.txt so oedisi build fails for the regular scenario.

Fixed. Plus I switched to a nested requirements.txt

>     * the scenario file includes an "existing_feeder_file" which doesn't have a fallback by the feeder federate if it doesn't exist.

I changed this to downloading instead. I do want to warn that the PV power in the gadal_ieee123 right now is still messed up.

>     * docker compose up doesn't work from the project directory and needs to be called from build to work. It should default to build and only need specification if you build to a different folder.

I don't really have a solution to this since we sort of need the yaml file to be with the build folders

>     * There is a PORT environment variable that causes the oedisi broker container to fail.
> 
> 
> <img alt="oedisi-multi-error" width="839" src="https://private-user-images.githubusercontent.com/20763042/304850339-56bcfbba-41ea-4a11-a50f-85b0b3d65828.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NDYxODQsIm5iZiI6MTcwNzk0NTg4NCwicGF0aCI6Ii8yMDc2MzA0Mi8zMDQ4NTAzMzktNTZiY2ZiYmEtNDFlYS00YTExLWE1MGYtODViMGIzZDY1ODI4LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDIxMjQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlODkwNzI3YTQ1Nzg2OTg0NjE1ZTQyMzI3OGI3NjYwNTc1ZWVkODk3OTdhMzEwN2MwYmUwNzlkOTMxNDI0NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PpUneChAQPJLhtO9inpnq4o0nFwXYfgvPpL1bpKEB2o">

I think this is part of Aadil's updates on his image. Using --pull=never should probably work, and I think --build on docker compose up should work too.

josephmckinsey commented:
@AadilLatif I just copied all that code from https://github.com/pnnl/oedisi_dopf, so I'm not sure about fixing it up...

josephmckinsey commented:
To myself: add a CI test.

AadilLatif commented on lindistflow_federate/lindistflow.py:
```diff
@@ -0,0 +1,944 @@
+            )
+            pcb, qcb = (
+                -(-z[1, 2][0] - math.sqrt(3) * z[1, 2][1]),
+                -(-z[0, 2][1] + math.sqrt(3) * z[1, 2][0]),
```
index is incorrect
